.github/workflows/update-gradle-wrapper.js

```diff
@@ -0,0 +1,112 @@
+const fs = require('fs');
+const https = require('https');
+
+async function fetchGradleVersions() {
+    return new Promise((resolve, reject) => {
+        https.get('https://services.gradle.org/versions/all', (res) => {
+            let data = '';
+            res.on('data', (chunk) => data += chunk);
+            res.on('end', () => resolve(JSON.parse(data)));
+        }).on('error', reject);
+    });
+}
+
+async function cleanup() {
+    try {
+        const { execSync } = require('child_process');
+        
+        const branches = execSync('git branch -r', { encoding: 'utf8' })
+            .split('\n')
+            .map(b => b.trim())
+            .filter(b => b.startsWith('origin/bot-upgradle-to-latest-wrapper-'));
+            
+        const sevenDaysAgo = new Date();
+        sevenDaysAgo.setDate(sevenDaysAgo.getDate() - 7);
+        
+        branches.forEach(branch => {
+            try {
+                const branchName = branch.replace('origin/', '');
+                const timestamp = branchName.split('-').pop();
+                const branchDate = new Date(
+                    timestamp.substring(0, 4),  // year
+                    parseInt(timestamp.substring(4, 6)) - 1,  // month (0-based)
+                    timestamp.substring(6, 8),
+                    timestamp.substring(8, 10),
+                    timestamp.substring(10, 12),
+                    timestamp.substring(12, 14)
+                );
+                
+                if (branchDate < sevenDaysAgo) {
+                    console.log(`Deleting old branch: ${branchName}`);
+                    execSync(`git push origin --delete ${branchName}`, { stdio: 'inherit' });
+                }
+            } catch (err) {
+                console.warn(`Warning: Failed to process branch ${branch}:`, err.message);
+            }
+        });
+    } catch (err) {
+        console.warn('Warning: Cleanup operation failed:', err.message);
+        // Don't throw, let main() continue
+    }
+}
+
+async function main() {
```

Comment:
❓ could we replace all of this with `./gradlew wrapper --gradle-version=nightly && ./gradlew wrapper` and then checking if there is a diff?